lib/puppet/type.rb

```diff
@@ -1316,6 +1318,9 @@ def pathbuilder
       and the second run will log the edit made by Puppet.)"
 
     validate do |list|
+      if Puppet.settings[:strict] != :off
+        Puppet.deprecation_warning(_("You are attempting to use the `audit` metaparameter in %{resource}. This parameter is deprecated and will be ignored in a future release.") % { resource: @resource.path })
```

Comment:
Is there a way to say more specifically where this warning was triggered? Wasn't able to find what I was looking for poking around in the resource object with `pry`...